### ISSUE TITLE
Base now exports `@main`

### ISSUE THREAD

**Issue by baggepinnen** [comonicon/Comonicon.jl#257]
Recently, [Base julia has started to export a `@main` macro](https://github.com/JuliaLang/julia/pull/51435), leading to conflicts with `Comonicon.@main`. It would be enough to edit all documentation of this package to qualify the use of `@main` with `Comonicon.@main`, and likely also a good idea to stop exporting `@main` so that the warning of conflict is silenced.

**Comment by Roger-luo** [comment]
yeah unfortunately that's gonna be a lot work that I don't current have the bandwidth to work on. some help would be nice!

**Comment by camilogarciabotero** [comment]
Just a couple of ideas that came to my mind:

```julia
@app
@run
@execute
@command
@compose
@runner
@launch
@commonicon
```

The `@app` is used for instance in [Typer](https://typer.tiangolo.com).

**Comment by Roger-luo** [comment]
Thanks! I like the first two (`@app` and `@run`) wondering if other people have an opinion on this. This would surely be breaking, so we will need to release a 2.0 version to have this. The non-breaking part probably can start with not exporting `@main` and documenting this behavior and tag a minor version? (If this is acceptable). 

Maybe I should release a patch version that caps the Julia version first so we make sure previous CLI apps don't get broken by default? Then we jump into a minor version document the `@main` macro, then collect a bit more breaking features to release 2.0? I'd like to change a few other things before doing the next major release, given that 1.0 of Comonicon wasn't around for very long...

**Comment by VPetukhov** [comment]
Perhaps, the quickest fix would be to update the documentation, replacing `@main` there with `Comonicon.@main`? It would still keep the warning, but at least the code would be working.

**Comment by Roger-luo** [comment]
yeah that would be nice!
